modules/dists/CyclicDist.chpl

```diff
@@ -1452,3 +1453,68 @@ proc newCyclicDom(rng: range...) {
 proc newCyclicArr(rng: range..., type eltType) {
   return newCyclicArr({(...rng)}, eltType);
 }
+
+
+proc CyclicArr.canDoOptimizedSwap(other) {
+  var domsMatch = true;
+
+  if this.dom != other.dom { // no need to check if this is true
+    if domsMatch {
```

Comment:
This will always be `true`, right? Is this conditional necessary?

Reply:
Hmm, good point.  TBH, I copy-pasted this from the Block swap routine without much thought, which seems to have the same property.  @e-kayrakli, looks like you added this pattern in https://github.com/chapel-lang/chapel/pull/16620.  Are you seeing anything we're missing?

Reply:
Yeah, that looks strange. Probably a leftover from an earlier implementation of some kind. Could you drop the equivalent in the Block implementation while here, or do you want me to?

Reply:
Yep, I've already got it, was just waiting for confirmation and a quick test before committing.  Thanks for taking a look!